new.go

```diff
@@ -232,6 +231,27 @@ func resolveImage(ctx context.Context, systemContext *types.SystemContext, store
 	}
 }
 
+func containerNameExist(name string, containers []storage.Container) bool {
+	for _, container := range containers {
+		for _, cname := range container.Names {
+			if cname == name {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
The lookup is quite expensive with O(n). It comes from storage, but I want to drop the idea here. Maybe, we can come up with a data type (in storage) that abstracts from such tasks, e.g.,
```golang
type Containers struct {
    ctr_slice []container
    ctr_map[string[container
}
```

Reply:
But certainly not in this PR. I just wanted to use this example to kick off a short discussion.

Reply:
I'd love to have a map available to lookup containers from, I'm just not sure how we create one efficiently that will be used enough times to justify the cost.  But a good discussion to have in case there's a way to do so.